Archive/Wiedijk100Theorems/SolutionOfCubic.lean

```diff
@@ -160,10 +154,9 @@ theorem cubic_eq_zero_iff_of_p_eq_zero (ha : a ≠ 0) (hω : IsPrimitiveRoot ω
         rw [hω.pow_eq_one, cube_root_of_unity_sum hω]; simp
       _ = (x - s) * (x - s * ω) * (x - s * ω ^ 2) := by ring
   rw [h₁, h₂, h₃, h₄ (x + b / (3 * a))]
-  ring_nf
+  ring_nf; trivial
```

Comment:
Did the behaviour of `ring_nf` change here? Should we fix this?

Reply:
This could be a regression in `ring_nf`. I made a smaller example:

```lean
example (a b : ℤ) : a+b=b+a ↔ b+a=a+b := by
  ring_nf -- works

example (a b : ℤ) : a+b=b+a ↔ b+a=a+b := by
  have : 3 = 3 := rfl
  ring_nf -- reduced to `True`
  trivial
```

Lean 3 doesn't have this `True` problem:

```lean
example (a b : ℤ) : a+b=b+a ↔ b+a=a+b := begin
  ring_nf,
end
example (a b : ℤ) : a+b=b+a ↔ b+a=a+b := begin
  have : 3 = 3 := rfl,
  ring_nf,
end
```

For now, I added a porting note.

Reply:
Fixed in #5246. `trivial` is no longer needed.